src/naming.jl

```diff
@@ -89,14 +89,16 @@ function savename(prefix::String, c, suffix::String;
         prefix = joinpath(prefix, dpre)
     end
 
-    visibles = setdiff(accesses, ignores)
-    labels = vecstring(visibles) # make it vector of strings
-    p = sortperm(labels)
+    # Perform access and ignore logic and sort
+    access_labels, ignore_labels = vecstring.((accesses, ignores))  # make it vector of strings
+    label_acc_tuples = collect(zip(access_labels, accesses))        # create tuple
+    filter!(t -> !(t[1] in ignore_labels), label_acc_tuples)        # ignores overwrites accesses
```

Comment:
`setdiff` should be used here instead. Not only it combine these 2 lines into one, it is also more performant.

Reply:
```

julia> a = ["a", "b", "c"]
3-element Array{String,1}:
 "a"
 "b"
 "c"

julia> b = ["b"]
1-element Array{String,1}:
 "b"

julia> setdiff(a, b)
2-element Array{String,1}:
 "a"
 "c"
```

Reply:
I understand how `setdiff` works. However, what we have now is a list of tuples.
```
julia> a_sym = [("a", :a),  ("b", :b)]; # or

julia> a_str = [("a", "a"), ("b", "b")];

julia> b_sym = [("b", :b)]; # or

julia> b_str = [("b", "b")];

julia> setdiff(a_sym, b_sym)
1-element Array{Tuple{String,Symbol},1}:
 ("a", :a)

julia> setdiff(a_sym, b_str)
2-element Array{Tuple{String,Symbol},1}:
 ("a", :a)
 ("b", :b)

julia> setdiff(a_str, b_sym)
2-element Array{Tuple{String,String},1}:
 ("a", "a")
 ("b", "b")

julia> setdiff(a_str, b_str)
1-element Array{Tuple{String,String},1}:
 ("a", "a")
```

The current implementation can deal with both cases.

Besides, I think performance should not be a concern here.

Reply:
I must admit I am extremely confused by this comment. Why would there be a datastructure like `a_sym = [("a", :a),  ("b", :b)];`???

`allaccess` and `allignore` both return a vector of keys to access the container. These are either vectors of symbols or vectors of strings, or the empty tuple `()`. In any case `setdiff` will always return the same, simple type as `accesses`.

I now see you have increased the complexity of the source without reason. Unless I am missing something, what you have to do is simply:
```julia
accesses = setdiff(accesses, ingores) # type stable operation, even if `ignores = ()`
```
just before the line `labels = vecstring(accesses)`. Then _change nothing of the subsequent code_. @sebastianpech am I missing something here? Because I think its that simple.

> Besides, I think performance should not be a concern here.

Well, on one hand we have a version that does (a) only a single line modification of the function's core, (b) is more performant and (c) is simpler. The best version is the version that should be implemented. Wouldn't you agree?

Reply:
Please read [my comment on the other conversation](https://github.com/JuliaDynamics/DrWatson.jl/pull/123#discussion_r376717544) on why simply swapping the lines as suggested by @sebastianpech doesn't work.